pkg/blackboxexporter/blackboxexporter.go

```diff
@@ -25,17 +25,30 @@ type BlackBoxExporter struct {
 	Ctx            context.Context
 	Image          string
 	NamespacedName types.NamespacedName
+	NodeSelector   corev1.NodeSelectorTerm
 }
 
 func New(client client.Client, log logr.Logger, ctx context.Context, blackBoxImage string, blackBoxExporterNamespace string) *BlackBoxExporter {
 	blackboxNamespacedName := types.NamespacedName{Name: blackboxexporter.BlackBoxExporterName, Namespace: blackBoxExporterNamespace}
-	return &BlackBoxExporter{client, log, ctx, blackBoxImage, blackboxNamespacedName}
+	nodeSelector := corev1.NodeSelectorTerm{
+		MatchExpressions: []corev1.NodeSelectorRequirement{
+			{
+				Key:      "node-role.kubernetes.io/infra",
+				Operator: corev1.NodeSelectorOpExists,
+			},
+		},
+	}
```

Comment:
I'm ok with this as a step, but we can release it this way first, then update the CRDs so that the desired nodeSelector is there explicitly (whether it's infra or master) and then remove this defaulting code